src/plugins/objectMigration/Migrations.js

```diff
@@ -224,6 +224,8 @@ define([
                                 .then(object => {
                                     telemetryObjects[element.id] = object;
                                 });
+                        } else {
+                            return false;
```

Comment:
The linter was complaining about the lack of a return in the case the if condition was not met. Adding an `else { return false }` made it happy. Would it be more correct to have it `return []` for this case?

Reply:
I think the issue is that the else you are adding is instead of returning a promise. So your promises array might end up looking like this, [promise1, promise2, false, promise3....].

Reply:
I'm not sure what is the proper way to resolve it. What should happen if one of the items in the elements array doesn't have an id?

Reply:
It looks the call after waiting for `promises` to resolve does not directly use all of the resolutions of the `promises`. It just makes sure the good promises resolve before doing something else. So the actual value of the return doesn't matter so much.

I don't really know. `map` is the easier way to code and understand this but not if we're going to be adding junk code to get around linting.... A `forEach` that pushes the good promises into the `promises` array would probably be best.

Reply:
Changed it to a forEach, but not sure how I can test it. Can't really see how objectMigration plugin is used.

Reply:
Changed this back to original change of just adding the return false when the element.id condition is not met. @akhenry - can you weigh in please?

Reply:
> Changed this back to original change of just adding the return false when the element.id condition is not met. @akhenry - can you weigh in please?

Reply:
JS will automatically wrap a primitive type in a Promise in this case, but it's better to be explicit. We can be explicit here by using [`Promise.resolve()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve) to wrap the `false` value.